src/sa.h

```diff
@@ -0,0 +1,112 @@
+#include <stdio.h>
+#include <string.h>
+#include <malloc.h>
+#include "parsers/simple-fasta-parser.h"
+#include "parsers/simple-fastq-parser.h"
+#include <stdint-gcc.h>
+#include "helper.h"
+
+int *constructSARadix(struct Fasta fasta) {
```

Comment:
Putting functions in header files is a dangerous game. The linker will kill you if you include the same file twice.

You can make the `static` (then you get duplicated object code, but the linker doesn't complain), but the sane thing to do is to put the functions in .c files and their prototypes in .h files.

Reply:
Good to know